File: cli/src/format.test.ts
import { describe, expect, test } from 'bun:test';
import { detectFormat } from './format.js';

describe('detectFormat', () => {
  test('detects markdown', () => {
    expect(detectFormat('docs/guide.md')).toBe('markdown');
    expect(detectFormat('readme.markdown')).toBe('markdown');
  });

  test('detects json/yaml/po/text', () => {
    expect(detectFormat('data.json')).toBe('json');
    expect(detectFormat('config.yaml')).toBe('yaml');
    expect(detectFormat('messages.po')).toBe('po');
    expect(detectFormat('notes.txt')).toBe('text');
  });
});

File: cli/internal/glossia/format_test.go
package glossia

import "testing"

func TestDetectFormat(t *testing.T) {
	if got := detectFormat("docs/guide.md"); got != FormatMarkdown {
		t.Fatalf("expected markdown, got %q", got)
	}
	if got := detectFormat("readme.markdown"); got != FormatMarkdown {
		t.Fatalf("expected markdown, got %q", got)
	}

	cases := map[string]Format{
		"data.json":     FormatJSON,
		"config.yaml":   FormatYAML,
		"messages.po":   FormatPO,
		"notes.txt":     FormatText,
		"unknown.file":  FormatText,
		"file.POT":      FormatPO,
		"file.YML":      FormatYAML,
		"file.MARKDOWN": FormatMarkdown,
	}

	for input, want := range cases {
		if got := detectFormat(input); got != want {
			t.Fatalf("detectFormat(%q) expected %q, got %q", input, want, got)
		}
	}
}
